projects/solar/test/test_fan_control_solar.c

```diff
@@ -0,0 +1,342 @@
+  fault_handler_init(&s_settings);
+
+  // Register Handlers to receive and unpack CAN message into static variables
+  can_register_rx_handler(SYSTEM_CAN_MESSAGE_SOLAR_FAULT_6_MPPTS, prv_ee_solar_fault_rx_handler,
+                          NULL);
+
+  // Used to reinitialize static testing variable GpioState s_gpio_state (default inactive high)
+  s_gpio_state = GPIO_STATE_HIGH;
+}
+
+void teardown_test(void) {}
+
+// Test that "fan fail" faults are raised properly
+void test_fan_fail_fault_handling(void) {
+  FanControlSolarSettings settings = {
+    .overtemp_addr = s_overtemp_addr,
+    .fan_fail_addr = s_fan_fail_addr,
+    .full_speed_addr = s_full_speed_addr,
+    .full_speed_temp_threshold_dC = s_full_speed_temp_threshold,
+    .mppt_count = s_mppt_count,
+  };
+
+  TEST_ASSERT_OK(fan_control_init(&settings));
+
+  // Test that there are no faults initially
+  MS_TEST_HELPER_ASSERT_NO_EVENT_RAISED();
+
+  // Test that fan fail fault is raised correctly when fan fail pin is interrupted
+  gpio_it_trigger_interrupt(&s_fan_fail_addr);
+
+  MS_TEST_HELPER_CAN_TX_RX(TEST_CAN_SOLAR_FAN_EVENT_TX, TEST_CAN_SOLAR_FAN_EVENT_RX);
+
+  TEST_ASSERT_EQUAL_MESSAGE(EE_SOLAR_FAULT_FAN_FAIL, s_ee_solar_fault,
+                            "Was expecting EE_SOLAR_FAULT_FAN_FAIL instead");
+  TEST_ASSERT_EQUAL_INT8(0, s_fault_data);
+}
+
+// Test that "overtemperature" faults are raised properly
+void test_overtemp_fault_handling(void) {
+  FanControlSolarSettings settings = {
+    .overtemp_addr = s_overtemp_addr,
+    .fan_fail_addr = s_fan_fail_addr,
+    .full_speed_addr = s_full_speed_addr,
+    .full_speed_temp_threshold_dC = s_full_speed_temp_threshold,
+    .mppt_count = s_mppt_count,
+  };
+
+  TEST_ASSERT_OK(fan_control_init(&settings));
+
+  // Test that there are no faults initially
+  MS_TEST_HELPER_ASSERT_NO_EVENT_RAISED();
+
+  // Test that fan fail fault is raised correctly when fan fail pin is interrupted
+  gpio_it_trigger_interrupt(&s_overtemp_addr);
+
+  MS_TEST_HELPER_CAN_TX_RX(TEST_CAN_SOLAR_FAN_EVENT_TX, TEST_CAN_SOLAR_FAN_EVENT_RX);
+
+  TEST_ASSERT_EQUAL_MESSAGE(EE_SOLAR_FAULT_FAN_OVERTEMPERATURE, s_ee_solar_fault,
+                            "Was expecting EE_SOLAR_FAULT_FAN_OVERTEMPERATURE instead");
+  TEST_ASSERT_EQUAL_INT8(0, s_fault_data);
+}
+
+// Test that full speed pins are raised correctly
+// Case 1: Temperature below threshold for all MPPTs
+void test_fan_overtemp_fault_fullspeed_handling_case_belowtemp(void) {
+  FanControlSolarSettings settings = {
+    .overtemp_addr = s_overtemp_addr,
+    .fan_fail_addr = s_fan_fail_addr,
+    .full_speed_addr = s_full_speed_addr,
+    .full_speed_temp_threshold_dC = s_full_speed_temp_threshold,
+    .mppt_count = s_mppt_count,
+  };
+
+  // Ensure fan_control_init works
+  TEST_ASSERT_OK(fan_control_init(&settings));
+
+  // Initialize all mppt statuses to nominal
+  for (int i = 0; i < 6; i++) {
+    data_store_set(DATA_POINT_MPPT_STATUS(i), SPV1020_NOMINAL_MASK);
+  }
```

Comment:
You shouldn't have to do this (here and in all the other tests) once you implement checking whether the mppt status is set in the data store before accessing it. `data_store_init` resets all the is-set flags to false, so you don't need to manually reset the statuses.